hyper/http11/connection.py

```diff
@@ -80,20 +84,24 @@ def __init__(self, host, port=None, secure=None, ssl_context=None,
         self._send_http_upgrade = not self.secure
         self._enable_push = kwargs.get('enable_push')
 
+        # CONNECT request (to create a tunnel via proxy) has a few special
+        # cases which needs to be processed differently.
+        self._is_connect_request = False
```

Comment:
So, I don't think this feels right. A connection is not a request: a request is just one of many things that can happen on a connection. In this case, all of the places it's used except for the response have access to the request verb, so they can just check what the request verb is. The better solution is to *tell* the response what the request verb was, which is good practice anyway.